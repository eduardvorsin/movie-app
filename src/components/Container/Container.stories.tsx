import type { Meta, StoryObj } from '@storybook/react';
import Container from './Container';

const meta: Meta<typeof Container> = {
	title: 'components/Container',
	tags: ['autodocs'],
	argTypes: {
		className: {
			description: 'Additional classes for the Container'
		},
		testId: {
			description: 'Id for testing the component',
		},
		children: {
			control: false,
			description: 'The component that will be wrapped in a container',
		},
		component: {
			control: false,
			description: 'The element that will serve as a container',
		}
	},
	component: Container,
	parameters: {
		docs: {
			description: {
				component: 'A component that serves as a bounding container when adapting pages',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof Container>;

export const Default: Story = {
	args: {
		className: '',
		testId: '',
		children: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Deleniti dolore fugit expedita mollitia laboriosam, ut explicabo natus quas distinctio, cum dolorem sit est autem quaerat. Itaque enim vel accusamus quod! Pariatur, ratione! At nemo aut expedita suscipit obcaecati voluptatum voluptates enim dolorum autem accusantium sunt fuga facilis, ratione voluptatibus cupiditate itaque, placeat illo accusamus doloremque veniam et sint quisquam tempora!'
	},
};