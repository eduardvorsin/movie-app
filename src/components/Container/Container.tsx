import { GeneralProps } from '@/types/shared';
import { ElementType, ReactNode } from 'react';

type Props = {
	children: ReactNode,
	component?: ElementType,
} & GeneralProps;

export default function Container({
	className,
	testId,
	component,
	children,
	...props
}: Props) {
	const Container = component ?? 'div';
	const classes = [
		'max-w-[77rem] mx-auto px-4',
		className,
	].join(' ');

	return (
		<Container
			className={classes}
			data-testid={testId}
			{...props}
		>
			{children}
		</Container>
	);
}