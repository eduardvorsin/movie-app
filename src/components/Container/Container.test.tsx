import { render, screen } from "@testing-library/react";
import Container from "./Container";

describe('Container tests', () => {
	it('is rendered correctly', () => {
		render(
			<Container testId='test-container'>test content</Container>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-container')).toBeInTheDocument();
	});

	it('is a basic snapshot', () => {
		render(
			<Container testId='test-container'>test content</Container>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-container')).toMatchSnapshot();
	});
});