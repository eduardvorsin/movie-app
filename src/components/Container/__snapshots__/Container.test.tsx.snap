// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container tests is a basic snapshot 1`] = `
<div
  class="max-w-[77rem] mx-auto px-4 "
  data-testid="test-container"
>
  test content
</div>
`;
